### PR TITLE
Create Settings Page

### PR DESCRIPTION
**Pull request by zachseidner1**
- Add generic toggle component for use anywhere in the application
- Finish settings page with networking
- Delete account is assumed to work once backend is fixed (currently facing internal server error)

### CODE REVIEW

File: src/pages/UserSettings/UserSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+  // Updating email alerts setting
+  const updateEmailAlerts = useMutation({
+    // Only way to have optional parameters for mutationFn
+    // https://github.com/TanStack/query/issues/4264#issuecomment-1268054812
+    mutationFn: (enabled: boolean) =>
+      api
+        .post<SettingShort>("/user/settings/", {
+          key: "notify",
+          value: emailTime,
```
Comment (jonathanjma) [inline src/pages/UserSettings/UserSettings.tsx:69]: does this incorporate the timezone? it seems like this just sends the time (ex. 15:30) but I think the backend code expects a timezone? https://github.com/jonathanjma/HappinessApp/blob/9605b958649c50178cfeffacc0c1e03b10ceafa8/happiness-backend/jobs/jobs.py#L146

File: src/pages/UserSettings/UserSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        )}
+        <h4 className="text-gray-600">Account Settings</h4>
+        <TextField
+          value={email}
+          hint="Enter a new email address"
+          onChangeValue={setEmail}
+          label="Change email:"
+        />
+        {changeEmailState && (
+          <label className="font-normal">{changeEmailState}</label>
```
Comment (jonathanjma) [inline src/pages/UserSettings/UserSettings.tsx:149]: make the error text red?

File: src/pages/UserSettings/UserSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+      mutationFn: (username: string) =>
+        api.put("/user/info/", {
+          data_type: "username",
+          data: username,
+        }),
+      onError: () => {
+        setChangeUsernameState("Username already taken.");
+      },
+      onSuccess: () => {
+        setChangeUsernameState("Username updated. Refresh to see changes.");
```
Comment (jonathanjma) [inline src/pages/UserSettings/UserSettings.tsx:91]: maybe use a toast for this to make it more obvious?

File: src/pages/UserSettings/UserSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+          onClick={() => {
+            if (username) {
+              changeUsername(username);
+            } else {
+              setChangeUsernameState("Username must not be empty");
+            }
+          }}
+        />
+        {changeUsernameState && (
+          <label className="font-normal">{changeUsernameState}</label>
```
Comment (jonathanjma) [inline src/pages/UserSettings/UserSettings.tsx:182]: make the error text red? also this should probably be before the change username button right?

File: src/pages/UserSettings/UserSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+  const { isLoading: emailChangeLoading, mutate: changeEmail } = useMutation({
+    mutationFn: (email: string) => {
+      return api.put("/user/info/", {
+        data_type: "email",
+        data: email,
+      });
+    },
+    onSuccess: () => {
+      setChangeEmailState("Email successfully updated");
```
Comment (jonathanjma) [inline src/pages/UserSettings/UserSettings.tsx:103]: same thing should we do a toast for this as well?

File: src/pages/UserSettings/UserSettings.tsx [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    },
+    onError: () => {
+      setChangeEmailState("Email may already be taken.");
+    },
+  });
+
+  return (
```
Comment (jonathanjma) [inline src/pages/UserSettings/UserSettings.tsx:110]: I think you forgot to add change password?

File: src/pages/UserSettings/UserSettings.tsx [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+              {emailTimeNetworkingState}
+            </label>
+          </>
+        )}
+        <h4 className="text-gray-600">Account Settings</h4>
+        <TextField
+          value={email}
+          hint="Enter a new email address"
+          onChangeValue={setEmail}
+          label="Change email:"
```
Comment (jonathanjma) [inline src/pages/UserSettings/UserSettings.tsx:146]: super small thing but maybe make this just "Email"? Having "Change Email" twice looks/feels redundant

File: src/pages/UserSettings/UserSettings.tsx [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+            } else {
+              setChangeEmailState("Email not valid");
+            }
+          }}
+        />
+        <TextField
+          value={username}
+          hint="Enter a new username"
+          onChangeValue={setUsername}
+          label="Change username:"
```
Comment (jonathanjma) [inline src/pages/UserSettings/UserSettings.tsx:166]: same with email maybe just make this "Username"?

**Review by jonathanjma — CHANGES_REQUESTED** [review]
Nice work, left some comments with things I noticed.

File: src/pages/UserSettings/UserSettings.tsx (hunk 3 above)
Comment (zachseidner1) [reply src/pages/UserSettings/UserSettings.tsx:91]: I'm following the design, I considered a toast but I feel like this is pretty clear imo. We can ask design on Tuesday as well.

File: src/pages/UserSettings/UserSettings.tsx (hunk 5 above)
Comment (zachseidner1) [reply src/pages/UserSettings/UserSettings.tsx:103]: ^

File: src/pages/UserSettings/UserSettings.tsx (hunk 1 above)
Comment (zachseidner1) [reply src/pages/UserSettings/UserSettings.tsx:69]: good point, I will add this.

**Review by jonathanjma — APPROVED** [review]
W, nice work ‼️

**Pull request merged**